### PR TITLE
itest: expect failure on direct rfq peer btc invoices

### PR DESCRIPTION
**Pull request by GeorgeTsagk**
## Description

Previously we'd consider it acceptable to settle direct rfq peer
invoices, which included no rfq scid, with asset HTLCs. This behavior
has been updated on the tapd invoice manager and we no longer accept
asset HTLCs on invoices that do not expect assets. This commit updates
such payments in our itests to instead expect a failure.

Tapd dependency:  https://github.com/lightninglabs/taproot-assets/pull/1244
Lnd dependency: https://github.com/lightningnetwork/lnd/pull/9357

### CODE REVIEW

File: itest/litd_custom_channels_test.go [hunk 1]
```diff
@@ -482,14 +482,12 @@ func testCustomChannels(_ context.Context, net *NetworkHarness,
 	// a direct channel invoice payment with no RFQ SCID present in the
 	// invoice.
 	// ------------
-	paidAssetAmount := createAndPayNormalInvoice(
+	createAndPayNormalInvoice(
 		t.t, charlie, dave, dave, 20_000, assetID, withSmallShards(),
+		withFailure(lnrpc.Payment_FAILED, lnrpc.PaymentFailureReason_FAILURE_REASON_INCORRECT_PAYMENT_DETAILS),
```
Comment (guggero) [inline itest/litd_custom_channels_test.go:487]: lll, here and below. Can use `failureIncorrectDetails` that's defined in `assets_test.go` instead.

**Review by guggero — COMMENTED** [review]
itest seems to fail.

**Comment by GeorgeTsagk** [comment]
> itest seems to fail.

interesting, could be flaky, didn't hit this on local tests

**Comment by GeorgeTsagk** [comment]
also linter seems to fail here with 
https://github.com/lightninglabs/lightning-terminal/actions/runs/12275164256/job/34257220772?pr=916#step:5:293

passes locally

**Review by ffranr — COMMENTED** [review]
:+1:  need to remove the temp commit (waiting on dep I guess?), otherwise looks good.

**Review by guggero — COMMENTED** [review]
LGTM pending dependant PRs and nit mentioned in earlier review.

**Comment by guggero** [comment]
Can be updated to `lnd` commit `[04767fe](https://github.com/lightningnetwork/lnd/pull/9367/commits/04767fe78c43c89755776bdc94cf919f76cc1e91)` which will remain stable, even after tagging the RC.

**Comment by GeorgeTsagk** [comment]
Updated `go.mod` deps & rebased
Waiting for CI

**Comment by guggero** [comment]
Just FYI: The itests broke due to the update to the latest `tapd` version on the base branch (`update-to-lnd-18-4`), will take a look. So the CI on this will likely also fail.

**Pull request closed without merging**